cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/AuthConfig.java

```diff
@@ -17,38 +17,45 @@
 package io.cdap.cdap.proto.sourcecontrol;
 
 import java.util.Objects;
-import javax.annotation.Nullable;
 
 /**
- * Auth Configuration for the linked repository
+ * Auth Configuration for the linked repository.
  */
 public class AuthConfig {
 
   private final AuthType type;
-  private final String tokenName;
-  private final String username;
-
-  public AuthConfig(AuthType authType, String tokenName, @Nullable String username) {
+  private final PatConfig patConfig;
+  
+  /**
+   * Constructor for AuthConfig.
+   * 
+   * @param authType {@link AuthType} to use
+   * @param patConfig {@link PatConfig} the PAT config
+   */
+  public AuthConfig(AuthType authType, PatConfig patConfig) {
     this.type = authType;
-    this.tokenName = tokenName;
-    this.username = username;
+    this.patConfig = patConfig;
   }
 
-  public AuthType getType() {
-    return type;
+  /**
+   * Construct an AuthConfig object based on another AuthConfig.
+   * 
+   * @param authConfig {@link AuthConfig} to construct from
```

Comment:
checkstyle warning. Though I don't really understand why we need this constructor, can we remove it?

Reply:
removed